test/foundry/core/AlloUnit.tree

```diff
@@ -0,0 +1,287 @@
+Allo::initialize
+└── given upgrade version is correct
+    ├── it should call _initializeOwner
+    ├── it should call _updateRegistry
+    ├── it should call _updateTreasury
+    ├── it should call _updatePercentFee
+    ├── it should call _updateBaseFee
+    └── it should call _updateTrustedForwarder
+
+Allo::createPoolWithCustomStrategy
+├── when strategy is zero address
+│   └── it should revert
+└── when calling with proper params
+    ├── it should call _createPool
+    └── it should return poolId
+
+Allo::createPool
+├── when strategy is zero address
+│   └── it should revert
+└── when calling with proper params
+    ├── it should call _createPool
+    └── it should return poolId
+
+Allo::updatePoolMetadata
+└── given sender is manager of pool
+    ├── it should call _checkOnlyPoolManager
+    ├── it should update metadata
+    └── it should emit event
+
+Allo::updateRegistry
+├── when sender is not owner
+│   └── it should revert
+└── when sender is owner
+    └── it should call _updateRegistry
+
+Allo::updateTreasury
+├── when sender is not owner
+│   └── it should revert
+└── when sender is owner
+    └── it should call _updateTreasury
+
+Allo::updatePercentFee
+├── when sender is not owner
+│   └── it should revert
+└── when sender is owner
+    └── it should call _updatePercentFee
+
+Allo::updateBaseFee
+├── when sender is not owner
+│   └── it should revert
+└── when sender is owner
+    └── it should call _updateBaseFee
+
+Allo::updateTrustedForwarder
+├── when sender is not owner
+│   └── it should revert
+└── when sender is owner
+    └── it should call _updateTrustedForwarder
+
+Allo::addPoolManagers
+└── given sender is admin of poolId
+    ├── it should call _checkOnlyPoolAdmin
+    └── it should call _addPoolManager
+
+Allo::removePoolManagers
+└── given sender is admin of poolId
+    ├── it should call _checkOnlyPoolAdmin
+    └── it should call _revokeRole
+
+Allo::addPoolManagersInMultiplePools
+└── given sender is admin of all poolIds
+    └── it should call addPoolManagers
+
+Allo::removePoolManagersInMultiplePools
+└── given sender is admin of all poolIds
+    └── it should call removePoolManagers
+
+Allo::recoverFunds
+├── when sender is not owner
+│   └── it should revert
+└── when sender is owner
+    ├── when token is native
+    │   └── it should transfer native token
+    └── when token is not native
+        └── it should transfer token
```

Comment:
for the tests let's make sure we also check it transfers the correct amount

Reply:
Yes, here I'm assuming that we will add a `expectCall` with exact parameters.